erpnext/projects/doctype/timesheet/timesheet.js

```diff
@@ -57,10 +57,37 @@ frappe.ui.form.on("Timesheet", {
 	},
 
 	make_invoice: function(frm) {
-		frappe.model.open_mapped_doc({
-			method: "erpnext.projects.doctype.timesheet.timesheet.make_sales_invoice",
-			frm: frm
+		let dialog = new frappe.ui.Dialog({
+			title: __("For Item"),
+			fields: [
+				{"fieldtype": "Link", "label": __("Item Code"), "fieldname": "item_code", "options":"Item"},
+				{"fieldtype": "Link", "label": __("Customer"), "fieldname": "customer", "options":"Customer"},
+				{"fieldtype": "Button", "label": __("Make Sales Invoice"), "fieldname": "make_sales_invoice", "cssClass": "btn-primary"},
+			]
+		});
+
+		dialog.fields_dict.make_sales_invoice.$input.click(function() {
```

Comment:
@rohitwaghchaure ,

Instead of adding Custom button can we use Primary Button to create Sales Invoice
e.g. `dialog.set_primary_action(__("Make Sales Invocie"), function() {` ?

Reply:
Fixed!